cmake/loki_transform.cmake

```diff
@@ -295,59 +298,75 @@ function( loki_transform_target )
-        list( FILTER _target_sources EXCLUDE REGEX ${source} )
-    endforeach()
-
-    if( NOT _PAR_T_COPY_UNMODIFIED )
-        # Update the target source list
-        set_property( TARGET ${_PAR_T_TARGET} PROPERTY SOURCES ${_target_sources} )
-    else()
-        # Copy the unmodified source files to the build dir
-        set( _target_sources_copy "" )
-        foreach( source ${_target_sources} )
-            get_filename_component( _source_name ${source} NAME )
-            list( APPEND _target_sources_copy ${CMAKE_CURRENT_BINARY_DIR}/${_source_name} )
-            ecbuild_debug( "[Loki] copy: ${source} -> ${CMAKE_CURRENT_BINARY_DIR}/${_source_name}" )
+    set(_TARGETS_POSTFIX "")
+    list( LENGTH _PAR_T_TARGET TARGETS_LENGTH )
+    if (TARGETS_LENGTH GREATER 1)
+        foreach(_target ${_PAR_T_TARGET})
+	    # sanitize target name, e.g., replace '.' with '-' (same things happen within Loki)
+            string(REPLACE "." "_" _sanitized_target ${_target})
+            list( APPEND _TARGETS_POSTFIX "_${_sanitized_target}")
         endforeach()
-        file( COPY ${_target_sources} DESTINATION ${CMAKE_CURRENT_BINARY_DIR} )
+    endif()
 
-        # Mark the copied files as build-time generated
-        set_source_files_properties( ${_target_sources_copy} PROPERTIES GENERATED TRUE )
 
-        # Update the target source list
-        set_property( TARGET ${_PAR_T_TARGET} PROPERTY SOURCES ${_target_sources_copy} )
-    endif()
+    foreach(_target _postfix IN ZIP_LISTS _PAR_T_TARGET _TARGETS_POSTFIX)
+        ecbuild_debug( "[Loki] Loki ${_target} transform: ${LOKI_SOURCES_TO_TRANSFORM${_postfix}}")
+        ecbuild_debug( "[Loki] Loki ${_target} append: ${LOKI_SOURCES_TO_APPEND${_postfix}}" )
+        ecbuild_debug( "[Loki] Loki ${_target} remove: ${LOKI_SOURCES_TO_REMOVE${_postfix}}" )
 
-    if ( LOKI_APPEND_LENGTH GREATER 0 )
-        # Mark the generated stuff as build-time generated
-        set_source_files_properties( ${LOKI_SOURCES_TO_APPEND} PROPERTIES GENERATED TRUE )
+        # Exclude source files that Loki has re-generated.
+        # Note, this is done explicitly here because the HEADER_FILE_ONLY
+        # property is not always being honoured by CMake.
+        get_target_property( _target_sources ${_target} SOURCES )
+        foreach( source ${LOKI_SOURCES_TO_REMOVE${_postfix}} )
+            # get_property( source_deps SOURCE ${source} PROPERTY OBJECT_DEPENDS )
+            list( FILTER _target_sources EXCLUDE REGEX ${source} )
+        endforeach()
 
-        # Add the Loki-generated sources to our target
-        target_sources( ${_PAR_T_TARGET} PRIVATE ${LOKI_SOURCES_TO_APPEND} )
-    endif()
+        if( NOT _PAR_T_COPY_UNMODIFIED )
+            # Update the target source list
+            set_property( TARGET ${_target} PROPERTY SOURCES ${_target_sources} )
+        else()
+            # Copy the unmodified source files to the build dir
```

Comment:
Are we actually using this feature (copying unmodified files) anywhere? Otherwise we could consider removing this in a future PR.

Reply:
Good question! I don't think so, but I am not sure.